### PR TITLE
singularity-eos: add v1.7.0, cleanup package

### PR DESCRIPTION
**Pull request by rbberger**

### CODE REVIEW

File: var/spack/repos/builtin/packages/singularity-eos/package.py [hunk 1]
```diff
@@ -130,25 +115,23 @@ def cmake_args(self):
...
             ),
             self.define(
                 "SINGULARITY_TEST_STELLARCOLLAPSE2SPINER",
-                ("stellarcollapse" in self.spec.variants["build_extra"] and "tests" in self.spec),
-            ),
-            self.define(
-                "SINGULARITY_TEST_PYTHON", ("python" in self.spec and "tests" in self.spec)
+                ("stellarcollapse" in self.spec.variants["build_extra"] and self.run_tests),
             ),
+            self.define("SINGULARITY_TEST_PYTHON", ("python" in self.spec and self.run_tests)),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/singularity-eos/package.py]: Should this be ensuring the `python` variant is enabled?

```suggestion
            self.define("SINGULARITY_TEST_PYTHON", ("+python" in self.spec and self.run_tests)),
```

File: var/spack/repos/builtin/packages/singularity-eos/package.py [hunk 2]
```diff
@@ -130,25 +115,23 @@ def cmake_args(self):
...
                 "SINGULARITY_BUILD_STELLARCOLLAPSE2SPINER",
                 "stellarcollapse" in self.spec.variants["build_extra"],
             ),
             self.define(
                 "SINGULARITY_TEST_STELLARCOLLAPSE2SPINER",
-                ("stellarcollapse" in self.spec.variants["build_extra"] and "tests" in self.spec),
-            ),
-            self.define(
-                "SINGULARITY_TEST_PYTHON", ("python" in self.spec and "tests" in self.spec)
+                ("stellarcollapse" in self.spec.variants["build_extra"] and self.run_tests),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/singularity-eos/package.py]: Shouldn't this be checking the `value` property of `variants`?

```suggestion
                ("stellarcollapse" in self.spec.variants["build_extra"].value and self.run_tests),
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256.  Just have a couple of variants-related questions.

File: var/spack/repos/builtin/packages/singularity-eos/package.py (hunk 1 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/singularity-eos/package.py]: Yes, good catch!

File: var/spack/repos/builtin/packages/singularity-eos/package.py (hunk 2 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/singularity-eos/package.py]: Double-checked this, and you're right. The old version seems to have "worked by mistake".

**Comment by rbberger** [comment]
@tldahlgren once the sha256 chaos has subsided, this should be good to go.

**Comment by rbberger** [comment]
@spackbot fix style

**Comment by rbberger** [comment]
@tldahlgren this should be ready to go. I just rebased in on top of the current develop.

**Review by tldahlgren — APPROVED** [review]
Re-confirmed sha256, rest LGTM

**Pull request merged**